feat(expo): Implement Google Sign-In support for Android and iOS

chriscanin opened the pull request:
## Description

This PR implements native Google Sign-In support for Expo iOS and Android apps using the `@react-native-google-signin/google-signin` library. This provides a seamless native authentication experience for users signing in with their Google accounts.

### What's New

- **New Hook**: `useSignInWithGoogle()` - Provides native Google Sign-In functionality for iOS and Android
  - Platform-specific implementations for iOS (`useSignInWithGoogle.ios.ts`) and Android (`useSignInWithGoogle.android.ts`)
  - Automatic handling of sign-in/sign-up transfer flow
  - Graceful error handling including user cancellation detection
  - Support for custom `unsafeMetadata` during authentication

### Key Features

- **Native UI**: Uses platform-native Google Sign-In UI instead of web-based OAuth flow
- **Automatic Flow Detection**: Seamlessly handles both new user sign-up and existing user sign-in
- **Error Handling**: Comprehensive error handling for:
  - Missing credentials
  - User cancellation (codes `SIGN_IN_CANCELLED` and `-5`)
  - Play Services availability (Android)
- **Environment Variable Support**: Reads credentials from both Expo config and environment variables
- **Full Test Coverage**: Complete unit tests with 100% coverage for the new hook

### Files Changed

- `packages/expo/src/hooks/useSignInWithGoogle.ts` - Cross-platform stub with helpful error message
- `packages/expo/src/hooks/useSignInWithGoogle.ios.ts` - iOS-specific implementation
- `packages/expo/src/hooks/useSignInWithGoogle.android.ts` - Android-specific implementation
- `packages/expo/src/hooks/__tests__/useSignInWithGoogle.test.ts` - Comprehensive unit tests
- `packages/expo/src/hooks/index.ts` - Export new hook
- `packages/expo/package.json` - Add optional peer dependency
- `packages/expo/vitest.setup.mts` - Add test mocks for expo-modules-core

### Implementation Details

**iOS Implementation:**
- Uses `expo-apple-authentication` for native Sign-In
- Supports `oauth_token_apple` strategy
- Handles identity token exchange

**Android Implementation:**
- Uses `@react-native-google-signin/google-signin` library
- Configures with Web Client ID and Android Client ID
- Checks Google Play Services availability
- Handles `google_one_tap` strategy with ID token

Both implementations:
- Automatically handle transfer flow when user doesn't exist
- Return consistent API: `{ createdSessionId, setActive, signIn, signUp }`
- Support optional `unsafeMetadata` parameter

### Breaking Changes

None - this is a new feature.

### Test Plan

1. Unit tests pass (`pnpm test --filter @clerk/clerk-expo`)
2. Build succeeds (`pnpm build`)
3. All 9 test cases pass including:
   - Successful sign-in for existing users
   - Transfer flow for new users
   - User cancellation handling (both error codes)
   - Play Services availability check (Android)
   - Missing ID token validation
   - Missing credentials validation
   - Clerk not loaded handling

### Documentation

Related documentation PR: [Link to clerk-docs PR]

## Checklist

- [x] `pnpm test` runs as expected.
- [x] `pnpm build` runs as expected.
- [x] JSDoc comments have been added for the new hook and all exported functions
- [x] Documentation has been created (see related PR in clerk-docs)

## Type of change

- [x] 🌟 New feature

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Native Google One Tap Sign-In for iOS and Android: configure, sign-in, create account, explicit sign-in, sign-out; iOS plugin can auto-configure URL scheme.
  * Cross-platform useSignInWithGoogle hook with shared flow and platform-specific adapters; JS helpers and public types for One Tap integration.

* **Tests**
  * Unit tests covering Android sign-in hook flows and edge cases.

* **Chores**
  * Packaging, build and module metadata updates; CocoaPods and Android build config added; .gitignore updated.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

chriscanin committed:
feat(expo): Implement Google Sign-In support for Android and iOS

chriscanin committed:
test(expo): Enhance mocks for expo-modules-core in test setup

chriscanin committed:
chore(package): remove local dependency on @clerk/clerk-expo

chriscanin committed:
feat(expo): Refactor Google Sign-In to use web-based OAuth flow for Android and enhance logging for iOS

chriscanin committed:
feat: add Clerk Google Sign-In module for Expo
- Removed package dependency for google sign in for android. Working on iOS next. Screw this persons package https://react-native-google-signin.github.io/docs. Going to go straight to the source.
- Implemented the Clerk Google Sign-In module for Android, including configuration and sign-in functionalities.
- Added necessary Gradle configuration and AndroidManifest setup.
- Created native Kotlin module with methods for configuring, signing in, creating accounts, and signing out.
- Developed TypeScript interface for the module, including response types for success, cancellation, and no saved credentials.
- Integrated the module with Expo's autolinking and created a config plugin for seamless integration.

chriscanin committed:
feat(expo): Add iOS support for Clerk Google Sign-In and enhance configuration options

chriscanin committed:
feat(expo): Implement iOS URL scheme configuration for Google Sign-In and update package version

chriscanin committed:
feat(google-signin): Add GoogleSignInActivityUnavailableException and improve error handling in sign-in flow

chriscanin committed:
feat(tests): Refactor Google Sign-In tests to use Clerk's One Tap Sign-In and improve error handling

chriscanin committed:
feat(google-signin): Add native Google Sign-In support for iOS and Android using built-in native modules

mikepitre reviewed (approved):
Swift code looks fine to me!

chriscanin commented:
!snapshot

clerk-cookie commented:
Hey @chriscanin - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/agent-toolkit | 0.2.6-snapshot.v20251205170211 |
| @clerk/astro | 2.16.4-snapshot.v20251205170211 |
| @clerk/backend | 2.25.0-snapshot.v20251205170211 |
| @clerk/chrome-extension | 2.8.7-snapshot.v20251205170211 |
| @clerk/clerk-js | 5.112.0-snapshot.v20251205170211 |
| @clerk/elements | 0.23.87-snapshot.v20251205170211 |
| @clerk/clerk-expo | 2.20.0-snapshot.v20251205170211 |
| @clerk/expo-passkeys | 0.4.23-snapshot.v20251205170211 |
| @clerk/express | 1.7.54-snapshot.v20251205170211 |
| @clerk/fastify | 2.6.6-snapshot.v20251205170211 |
| @clerk/localizations | 3.29.0-snapshot.v20251205170211 |
| @clerk/nextjs | 6.35.6-snapshot.v20251205170211 |
| @clerk/nuxt | 1.13.4-snapshot.v20251205170211 |
| @clerk/clerk-react | 5.57.1-snapshot.v20251205170211 |
| @clerk/react-router | 2.3.1-snapshot.v20251205170211 |
| @clerk/remix | 4.13.21-snapshot.v20251205170211 |
| @clerk/shared | 3.37.0-snapshot.v20251205170211 |
| @clerk/tanstack-react-start | 0.27.6-snapshot.v20251205170211 |
| @clerk/testing | 1.13.20-snapshot.v20251205170211 |
| @clerk/themes | 2.4.41-snapshot.v20251205170211 |
| @clerk/types | 4.101.4-snapshot.v20251205170211 |
| @clerk/vue | 1.17.1-snapshot.v20251205170211 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/agent-toolkit`
```sh
npm i @clerk/agent-toolkit@0.2.6-snapshot.v20251205170211 --save-exact
```
`@clerk/astro`
```sh
npm i @clerk/astro@2.16.4-snapshot.v20251205170211 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@2.25.0-snapshot.v20251205170211 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@2.8.7-snapshot.v20251205170211 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.112.0-snapshot.v20251205170211 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.23.87-snapshot.v20251205170211 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.20.0-snapshot.v20251205170211 --save-exact
```
`@clerk/expo-passkeys`
```sh
npm i @clerk/expo-passkeys@0.4.23-snapshot.v20251205170211 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.7.54-snapshot.v20251205170211 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@2.6.6-snapshot.v20251205170211 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.29.0-snapshot.v20251205170211 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@6.35.6-snapshot.v20251205170211 --save-exact
```
`@clerk/nuxt`
```sh
npm i @clerk/nuxt@1.13.4-snapshot.v20251205170211 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.57.1-snapshot.v20251205170211 --save-exact
```
`@clerk/react-router`
```sh
npm i @clerk/react-router@2.3.1-snapshot.v20251205170211 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.13.21-snapshot.v20251205170211 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@3.37.0-snapshot.v20251205170211 --save-exact
```
`@clerk/tanstack-react-start`
```sh
npm i @clerk/tanstack-react-start@0.27.6-snapshot.v20251205170211 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.13.20-snapshot.v20251205170211 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.4.41-snapshot.v20251205170211 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.101.4-snapshot.v20251205170211 --save-exact
```
`@clerk/vue`
```sh
npm i @clerk/vue@1.17.1-snapshot.v20251205170211 --save-exact
```

chriscanin committed:
fix(build.gradle): centralize dependency versions for easier updates

chriscanin committed:
refactor(types): Update imports from '@clerk/types' to '@clerk/shared/types' across multiple files

chriscanin committed:
fix(tests): update nonce value in useSignInWithGoogle test to use UUID format

chriscanin committed:
fix(deps): correct package name for @clerk/expo and add native Google Sign-In support for iOS and Android

chriscanin committed:
refactor(imports): update imports from '@clerk/clerk-react' to '@clerk/react/legacy' in multiple hooks and tests

dstaley commented on packages/expo/app.plugin.js:
```diff
@@ -0,0 +1,44 @@
+  return config;
+}
+
+module.exports = createRunOncePlugin(withClerkGoogleSignIn, '@clerk/clerk-expo', '1.0.0');
```
In core-3 the package is named `@clerk/expo`. Also we should probably use the version from package.json instead of it being hardcoded here.

dstaley commented on packages/expo/expo-module.config.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "platforms": ["android", "ios"],
```
not super familiar with native modules, but would this in any way prevent the Expo package from being used with Expo Web?

dstaley commented on packages/expo/package.json:
```diff
@@ -56,16 +56,28 @@
+      "types": "./dist/google-one-tap/index.d.ts",
+      "default": "./dist/google-one-tap/index.js"
+    },
+    "./app.plugin.js": "./app.plugin.js",
```
does this need to be in our `exports` to work correctly?

chriscanin replied:
I believe this is required (unless you know of a different way it could be done).  In the package exports node restricts access to only listed paths. Without the entry our require('@clerk/expo/app.plugin.js') fails with ERR_PACKAGE_PATH_NOT_EXPORTED. 

Expos plugin resolution needs the path accessible to load the config plugin

chriscanin commented on packages/expo/src/google-one-tap/ClerkGoogleOneTapSignIn.ts:
````diff
@@ -0,0 +1,205 @@
+ *
+ * @example
+ * ```typescript
+ * import { ClerkGoogleOneTapSignIn } from '@clerk/clerk-expo';
````
ClerkGoogleOneTapSignIn gives access to the native Google Sign-In module. Users would use this if they were making a custom flow. 

useSignInWithGoogle() is a hook built on top of clerkgoogleonetapsignin that handles the full authentication flow automatically

chriscanin commented on packages/expo/src/google-one-tap/ClerkGoogleOneTapSignIn.ts:
```diff
@@ -0,0 +1,205 @@
+ * });
+ *
+ * // Sign in with nonce
+ * const nonce = generateNonce();
```
Yes this is something the developer needs to generate themselves if they are using the clerkgoogleonetapsignin api directly. I will change the example to show use with expo crypto so that it is a bit more clear.

chriscanin committed:
feat(legacy): Add legacy module support and update nonce generation to use Crypto.randomUUID()

chriscanin committed:
Resolves:
https://github.com/clerk/javascript/pull/7208#discussion_r2607438952
and
https://github.com/clerk/javascript/pull/7208#discussion_r2607444588

chriscanin committed:
refactor(types): Simplify type exports for Google authentication hooks in Android and iOS

Resolves: https://github.com/clerk/javascript/pull/7208#discussion_r2607447248

chriscanin committed:
refactor(auth): streamline Google Sign-In logic by consolidating authentication flow into a shared function for Android and iOS

Resolves:
https://github.com/clerk/javascript/pull/7208#discussion_r2607452602
and
https://github.com/clerk/javascript/pull/7208#discussion_r2607457572
and
https://github.com/clerk/javascript/pull/7208#discussion_r2607459873

chriscanin committed:
refactor(expo): restructure plugin to match standard RN package pattern

  - Simplify app.plugin.js to one-liner
  - Add plugin/src/withClerkExpo.ts TypeScript source
  - Add plugin/tsconfig.json for compilation
  - Add build:plugin script to package.json

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
fix(tests): Update import statement in useSignInWithGoogle test to match module structure

chriscanin committed:
chore(dependencies): Update @expo/config-plugins, @expo/config-types, @expo/json-file, and @expo/plist versions in pnpm-lock.yaml

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

clerk-cookie commented:
Hey @chriscanin - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/agent-toolkit | 0.2.9-snapshot.v20251216175437 |
| @clerk/astro | 3.0.0-snapshot.v20251216175437 |
| @clerk/backend | 3.0.0-snapshot.v20251216175437 |
| @clerk/chrome-extension | 3.0.0-snapshot.v20251216175437 |
| @clerk/clerk-js | 3.0.0-snapshot.v20251216175437 |
| @clerk/dev-cli | 1.0.0-snapshot.v20251216175437 |
| @clerk/expo | 3.0.0-snapshot.v20251216175437 |
| @clerk/expo-passkeys | 1.0.0-snapshot.v20251216175437 |
| @clerk/express | 2.0.0-snapshot.v20251216175437 |
| @clerk/fastify | 2.6.9-snapshot.v20251216175437 |
| @clerk/localizations | 4.0.0-snapshot.v20251216175437 |
| @clerk/nextjs | 7.0.0-snapshot.v20251216175437 |
| @clerk/nuxt | 2.0.0-snapshot.v20251216175437 |
| @clerk/react | 6.0.0-snapshot.v20251216175437 |
| @clerk/react-router | 3.0.0-snapshot.v20251216175437 |
| @clerk/shared | 4.0.0-snapshot.v20251216175437 |
| @clerk/tanstack-react-start | 1.0.0-snapshot.v20251216175437 |
| @clerk/testing | 2.0.0-snapshot.v20251216175437 |
| @clerk/ui | 1.0.0-snapshot.v20251216175437 |
| @clerk/upgrade | 2.0.0-snapshot.v20251216175437 |
| @clerk/vue | 2.0.0-snapshot.v20251216175437 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/agent-toolkit`
```sh
npm i @clerk/agent-toolkit@0.2.9-snapshot.v20251216175437 --save-exact
```
`@clerk/astro`
```sh
npm i @clerk/astro@3.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@3.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@3.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@3.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/dev-cli`
```sh
npm i @clerk/dev-cli@1.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/expo`
```sh
npm i @clerk/expo@3.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/expo-passkeys`
```sh
npm i @clerk/expo-passkeys@1.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@2.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@2.6.9-snapshot.v20251216175437 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@4.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@7.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/nuxt`
```sh
npm i @clerk/nuxt@2.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/react`
```sh
npm i @clerk/react@6.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/react-router`
```sh
npm i @clerk/react-router@3.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@4.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/tanstack-react-start`
```sh
npm i @clerk/tanstack-react-start@1.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@2.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/ui`
```sh
npm i @clerk/ui@1.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/upgrade`
```sh
npm i @clerk/upgrade@2.0.0-snapshot.v20251216175437 --save-exact
```
`@clerk/vue`
```sh
npm i @clerk/vue@2.0.0-snapshot.v20251216175437 --save-exact
```

chriscanin committed:
chore(version): Bump @clerk/clerk-js version from 2.19.23 to 5.114.0

chriscanin committed:
chore(version): Downgrade @clerk/expo version from 2.19.26 to 2.19.10

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
fix(plugin): update package import path for @clerk/expo

Resolves: https://github.com/clerk/javascript/pull/7208/changes#r2620772406

chriscanin committed:
refactor(google-one-tap): update module description and usage guidance

- Changed the module description to clarify its internal status.
- Removed example usage and added a note to use the `useSignInWithGoogle` hook instead.
- Updated platform support to include both Android and iOS.

Resolves:
https://github.com/clerk/javascript/pull/7208/changes#r2608078644

chriscanin committed:
feat(google-one-tap): add new error code for Android activity unavailability

- Introduced 'E_ACTIVITY_UNAVAILABLE' to the GoogleSignInErrorCode type to handle cases where the Android activity is unavailable.
- Updated documentation to clarify the meaning of each error code.

Resolves:
https://github.com/clerk/javascript/pull/7208/changes#r2607408175

chriscanin committed:
refactor(google-one-tap): simplify Google Sign-In hooks for Android and iOS

- Replaced the existing `useSignInWithGoogle` implementation with a factory function `createUseSignInWithGoogle` to handle platform-specific configurations.
- Updated the Android and iOS hooks to utilize the new factory function, streamlining the authentication flow and reducing code duplication.

This change enhances maintainability and clarity in the Google Sign-In implementation across platforms.

Resolves: https://github.com/clerk/javascript/pull/7208/changes#r2607452602

chriscanin committed:
chore(package): remove redundant package.json entries

- Eliminated the duplicate entry for "./package.json" in the package.json file of the expo package, streamlining the file structure.

This change helps maintain a cleaner and more efficient package configuration.

Resolves:
https://github.com/clerk/javascript/pull/7208/changes#r2607473417

chriscanin commented on packages/expo/package.json:
```diff
@@ -56,16 +56,28 @@
+    "passkeys",
+    "secure-store",
+    "resource-cache",
+    "token-cache"
```
Some of these are unnecessary like the entries for web, local-credentials, passkeys, secure-store, resource-cache, and token-cache. I have now removed the unnecessary entries. 

The others are neccesary:
**dist** - All the compiled JavaScript/TypeScript code. Every export points here (./dist/passkeys/index.js, etc.). Without this, no JS code ships.

**android** - Native Kotlin code for the Google Sign-In module. Expo autolinking picks this up and compiles it into the Android app.

**ios** - Native Swift code for the Google Sign-In module. Same thing - Expo autolinking compiles it into the iOS app.

**expo-module.config.json** - Tells Expo "this package has native modules" and where to find them. Without this, Expo wouldn't autolink the native code.

**app.plugin.js** - Entry point for the Expo config plugin. When users add "plugins": ["@clerk/expo"] to app.json, Expo loads this to modify native project config (like adding URL schemes to Info.plist).

**plugin** - The compiled plugin code that app.plugin.js requires.

chriscanin committed:
chore(package): clean up package.json by removing unnecessary entries

- Removed several redundant entries from the package.json file in the expo package, retaining only the essential "plugin" entry. This simplifies the file and improves clarity.

resolves: https://github.com/clerk/javascript/pull/7208/changes#r2607474639

chriscanin committed:
chore(package): update peerDependencies in package.json for expo

- Added a new peer dependency for expo with version constraints ">=53 <55" to ensure compatibility with related packages.

This change helps maintain proper version alignment within the package ecosystem.

Resolves: https://github.com/clerk/javascript/pull/7208/changes#r2607477473

chriscanin committed:
feat(google-one-tap): enhance Google client ID retrieval in useSignInWithGoogle hook

- Introduced a new helper function `getGoogleClientIds` to dynamically import `expo-constants` and retrieve Google client IDs, falling back to environment variables if necessary.
- Updated the `createUseSignInWithGoogle` function to utilize the new helper for improved flexibility and maintainability.

This change streamlines the process of obtaining client IDs for Google Sign-In across different environments.

Resolves: https://github.com/clerk/javascript/pull/7208/changes#r2620772398

chriscanin committed:
fix(google-one-tap): enhance error handling in Google Sign-In

- Updated the `isErrorWithCode` function to check for both `code` and `message` properties in error objects, improving the robustness of error handling for Google Sign-In.

This change ensures that error responses are more informative and easier to debug.

Resolves: https://github.com/clerk/javascript/pull/7208/changes#r2624871828

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
chore(pnpm): update expo version in pnpm-lock.yaml

- Added expo version 54.0.23 with specified version constraints '>=53 <55' to ensure compatibility with related packages.
- Updated the engines field for '@expo/bunyan@4.0.1' to use a more explicit format.

This change helps maintain proper version alignment and clarity in the package ecosystem.

chriscanin commented:
!snapshot

dstaley commented on packages/expo/package.json:
```diff
@@ -59,17 +59,24 @@
-    "build": "tsup",
+    "build": "tsup && pnpm build:plugin",
     "build:declarations": "tsc -p tsconfig.declarations.json",
+    "build:plugin": "tsc -p plugin/tsconfig.json",
```
Can we reuse tsup instead of a separate tsconfig and build step?

chriscanin replied:
Yes I believe we can. I have adjusted the project structure to support this. Testing now.

chriscanin committed:
refactor(expo): update app.plugin.js path and clean up package.json

- Changed the export path in app.plugin.js to point to the new distribution directory.
- Removed the build:plugin script and unnecessary entries from package.json, streamlining the configuration.

These changes are in effort to continue using tsup.

Resolves:
https://github.com/clerk/javascript/pull/7208/changes#r2628292077
and
https://github.com/clerk/javascript/pull/7208/changes#r2628293763

chriscanin commented:
!snapshot

clerk-cookie commented:
Hey @chriscanin - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/agent-toolkit | 0.2.9-snapshot.v20251217194100 |
| @clerk/astro | 3.0.0-snapshot.v20251217194100 |
| @clerk/backend | 3.0.0-snapshot.v20251217194100 |
| @clerk/chrome-extension | 3.0.0-snapshot.v20251217194100 |
| @clerk/clerk-js | 6.0.0-snapshot.v20251217194100 |
| @clerk/dev-cli | 1.0.0-snapshot.v20251217194100 |
| @clerk/expo | 3.0.0-snapshot.v20251217194100 |
| @clerk/expo-passkeys | 1.0.0-snapshot.v20251217194100 |
| @clerk/express | 2.0.0-snapshot.v20251217194100 |
| @clerk/fastify | 2.6.9-snapshot.v20251217194100 |
| @clerk/localizations | 4.0.0-snapshot.v20251217194100 |
| @clerk/nextjs | 7.0.0-snapshot.v20251217194100 |
| @clerk/nuxt | 2.0.0-snapshot.v20251217194100 |
| @clerk/react | 6.0.0-snapshot.v20251217194100 |
| @clerk/react-router | 3.0.0-snapshot.v20251217194100 |
| @clerk/shared | 4.0.0-snapshot.v20251217194100 |
| @clerk/tanstack-react-start | 1.0.0-snapshot.v20251217194100 |
| @clerk/testing | 2.0.0-snapshot.v20251217194100 |
| @clerk/ui | 1.0.0-snapshot.v20251217194100 |
| @clerk/upgrade | 2.0.0-snapshot.v20251217194100 |
| @clerk/vue | 2.0.0-snapshot.v20251217194100 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/agent-toolkit`
```sh
npm i @clerk/agent-toolkit@0.2.9-snapshot.v20251217194100 --save-exact
```
`@clerk/astro`
```sh
npm i @clerk/astro@3.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@3.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@3.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@6.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/dev-cli`
```sh
npm i @clerk/dev-cli@1.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/expo`
```sh
npm i @clerk/expo@3.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/expo-passkeys`
```sh
npm i @clerk/expo-passkeys@1.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@2.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@2.6.9-snapshot.v20251217194100 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@4.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@7.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/nuxt`
```sh
npm i @clerk/nuxt@2.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/react`
```sh
npm i @clerk/react@6.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/react-router`
```sh
npm i @clerk/react-router@3.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@4.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/tanstack-react-start`
```sh
npm i @clerk/tanstack-react-start@1.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@2.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/ui`
```sh
npm i @clerk/ui@1.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/upgrade`
```sh
npm i @clerk/upgrade@2.0.0-snapshot.v20251217194100 --save-exact
```
`@clerk/vue`
```sh
npm i @clerk/vue@2.0.0-snapshot.v20251217194100 --save-exact
```

chriscanin committed:
Merge remote-tracking branch 'origin/main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
refactor(expo): update import statement for package.json in withClerkExpo plugin

- Changed the import of package.json from require to ES module syntax in order to fix the no "require" imports allowed eslint warning.

dstaley commented on packages/expo/src/hooks/useSignInWithGoogle.types.ts:
```diff
@@ -0,0 +1,14 @@
+import type { SetActive, SignInResource, SignUpResource } from '@clerk/shared/types';
+
+export type SignUpUnsafeMetadata = Record<string, unknown>;
```
I'm pretty sure you don't need to declare this, as we augment the global scope to support the customer defining their own type for this. I think you can just reference `SignUpUnsafeMetadata` directly without any imports or anything.

dstaley reviewed (approved):
one note about SignUpUnsafeMetadata, but otherwise LGTM!

chriscanin committed:
refactor(expo): remove unused SignUpUnsafeMetadata type from Google sign-in types

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin commented:
!snapshot

chriscanin committed:
chore(expo): add app.plugin.js entry to package.json exports

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
resolve merge conflicts.

chriscanin committed:
Merge branch 'main' into chris/mobile-289-expo-google-universal-sign-in

chriscanin committed:
test(expo): enhance useSignInWithGoogle tests with isClerkAPIResponseError mock
